### DISCUSSION TITLE
Adding a timer to an ETO form

### DISCUSSION THREAD

**Discussion by RayaanABT** [picoe/Eto#2083] [Q&A]
What is the most suitable method to add a timer, starting at zero and counting in seconds, to the ETO form?

Could this be added to the tutorials?
https://github.com/picoe/Eto/wiki/Tutorials

**Comment by cwensley** [comment]
You mean like a timer to display to the user?  A simple `Label` and a `UITimer` should do the trick..

```
var label = new Label();

// put the label somewhere in your UI

var start = DateTime.Now;
var timer = new UITimer { Interval = 1.0 };
timer.Elapsed += (sender, e) => {
  label.Text = $"Elapsed: {DateTime.Now - start}";
};
timer.Start();
```

**Reply by pha3z** [reply]
I think there are more interesting questions about the UI Timer:

What's special about the UITimer versus some other .Net timer?
Is the UITimer doing tricks to make sure the event it fires happens on the UI Thread?  If yes, is that all that's special about it?

**Reply by cwensley** [reply]
That is correct, the UI Timer fires on the UI thread, thus you can interact with the UI controls in its Elapsed event.  Other timers execute on a separate thread so any calls to the UI will fail unless you use `Application.Instance.Invoke/AsyncInvoke`. 

The UI timer is also typically scheduled on the UI run loop so it isn't necessarily as accurate as normal timers.
